mpc_testing_with_sealights.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+# making sure anything that fails during the script stops the rest of the script from running
+set -euo pipefail
+
+# adding some log-like lines for what's happening here
+exec > >(while IFS= read -r line; do echo "$(date +'%Y-%m-%d %H:%M:%S') $line"; done)
+exec 2>&1
+
+MODE=""
+PULL_REQUEST_NUMBER=""
+LATEST_COMMIT_SHA=""
+DATE=""
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    --mode)
+      MODE="$2"
+      shift 2
+      ;;
+    --pr-number)
+      PULL_REQUEST_NUMBER="$2"
+      shift 2
+      ;;
+    --commit-sha)
+      LATEST_COMMIT_SHA="$2"
+      shift 2
+      ;;
+    --date)
+      DATE="$2"
+      shift 2
+      ;;
+    *)
+      echo "Unknown option: $1" >&2
+      exit 1
+      ;;
+  esac
+done
+
+# Validate mode and required parameters
+if [[ -z "$MODE" ]]; then
+  echo "Error: --mode is required" >&2
+  exit 1
+fi
+
```

Comment:
I would separate downloading dependencies and running processes, so that GitHub Actions` cache can be used

Reply:
SeaLights are working on fetching their agent via go releases currently and I'm checking in on their progress on a daily basis. Using github actions cache would still be identical in both workflows and wouldn't be DRY if I separated them. I'm not sure we should invest in a different method for the moment, if a different solution is about to come out any day now. SeaLights are based in Israel and have been working through this holiday season so progress is being made.